lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -860,6 +860,20 @@ public synchronized void nack(long serial, int count, ErrorInfo reason) {
 				}
 			}
 		}
+
+		/**
+		 * reset the pending message queue, failing any currently pending messages.
+		 * Used when a resume fails and we get a different connection id.
+		 * @param oldMsgSerial the next message serial number for the old
+		 *		connection, and thus one more than the highest message serial
+		 *		in the queue.
+		 */
+		public synchronized void reset(long oldMsgSerial) {
```

Comment:
I think `reset()` should take an `ErrorInfo` argument and pass that to the nack, and then the specific error is determined by whoever calls `reset()`. There will be different situations that will require different error messages/codes.